diff --git a/pylib/gyp/generator/msvs.py b/pylib/gyp/generator/msvs.py
index 05783d6..539614c 100644
--- a/pylib/gyp/generator/msvs.py
+++ b/pylib/gyp/generator/msvs.py
@@ -1963,13 +1963,17 @@ def _AppendFiltersForMSBuild(parent_filter_name, sources,
     if isinstance(source, MSVSProject.Filter):
       # We have a sub-filter.  Create the name of that sub-filter.
       if not parent_filter_name:
-        filter_name = source.name
+        if source.name == '..':
+          filter_name = ''
+        else:
+          filter_name = source.name
       else:
         filter_name = '%s\\%s' % (parent_filter_name, source.name)
       # Add the filter to the group.
-      filter_group.append(
-          ['Filter', {'Include': filter_name},
-           ['UniqueIdentifier', MSVSNew.MakeGuid(source.name)]])
+      if filter_name:
+        filter_group.append(
+            ['Filter', {'Include': filter_name},
+             ['UniqueIdentifier', MSVSNew.MakeGuid(source.name)]])
       # Recurse and add its dependents.
       _AppendFiltersForMSBuild(filter_name, source.contents,
                                extension_to_rule_name,
